x/mongo/driver/crypt.go

```diff
@@ -43,24 +43,46 @@ type CryptOptions struct {
 	BypassAutoEncryption bool
 }
 
-// Crypt consumes the libmongocrypt.MongoCrypt type to iterate the mongocrypt state machine and perform encryption
+// Crypt is an interface implemented by types that can encrypt and decrypt instances of
+// bsoncore.Document.
+//
+// Users should rely on the driver's crypt type (used by default) for encryption and decryption
+// unless they are perfectly confident in another implementation of Crypt.
+type Crypt interface {
```

Comment:
Can you add a test with a type implementing `driver.Crypt` to test that automatic encryption and decryption work with the interface? That will give some assurance we call the interface as expected.

This is a WIP of what I have in mind: https://github.com/kevinAlbs/mongo-go-driver/commit/8e8b74adb924201f98b9f6f9b2420efad630c12c

Reply:
Thanks for the WIP, implementing some tests now.

Reply:
Looks good. I left a suggestion for additional assertions and exercising automatic decryption.